### PR TITLE
Switch stashing to key_value with fallback

### PR DESCRIPTION
**Pull request by Pchelolo**
Step 1 of the process - only switches the stashing.

Bug: [T215956](https://phabricator.wikimedia.org/T215956)

### CODE REVIEW

File: sys/parsoid.js [hunk 1]
```diff
@@ -224,7 +230,7 @@ class ParsoidService {
     }
 
     getFallbackBucketURI(rp, format, tid) {
-        return this.getStashBucketURI(rp, format, tid);
+        return this.getOldStashBucketURI(rp, format, tid);
```
Comment (d00rman) [inline sys/parsoid.js]: I think this is a relic from the transition to the complicated storage semantics. AFAICT, we don't need it any more, right?

File: sys/parsoid.js [hunk 2]
```diff
@@ -281,26 +287,6 @@ class ParsoidService {
     generateAndSave(hyper, req, format, currentContentRes) {
         // Try to generate HTML on the fly by calling Parsoid
         const rp = req.params;
-        // Helper for retrieving original content from storage & posting it to
```
Comment (d00rman) [inline sys/parsoid.js]: This was commented out during the last transition, but we forgot to start using it again. It helps Parsoid making a quicker conversion for the same revid, so we should probably keep this and activate it after the current transition.

Either way, this change does not belong in this PR

File: sys/parsoid.js [hunk 3]
```diff
@@ -510,23 +496,33 @@ class ParsoidService {
...
-
-        return P.props({
-            html: getStash('html'),
-            'data-parsoid': getStash('data-parsoid'),
-            wikitext: getStash('wikitext')
+        return hyper.get({
+            uri: this.getStashBucketURI(rp.domain, rp.title, rp.revision, tid)
         })
         .then((res) => {
+            res = JSON.parse(res.body.toString('utf8'));
```
Comment (d00rman) [inline sys/parsoid.js]: Let's be consistent and use `mwUtil.decodeBody()`

File: sys/parsoid.js [hunk 4]
```diff
@@ -630,29 +626,18 @@ class ParsoidService {
...
-                hyper.put({
-                    uri: this.getStashBucketURI(rp, 'data-parsoid', tid),
-                    headers: original.body['data-parsoid'].headers,
-                    body: original.body['data-parsoid'].body
-                }),
-                hyper.put({
-                    uri: this.getStashBucketURI(rp, 'wikitext', tid),
+        return transformPromise.then((original) => hyper.put({
+            uri: this.getStashBucketURI(rp.domain, rp.title, rp.revision, tid),
+            body: Buffer.from(JSON.stringify({
```
Comment (d00rman) [inline sys/parsoid.js]: Also for the sake of consistency, perhaps we should create a `mwUtils.encodeBody()` function that encodes this, and use it in the mobile sections part too.

File: sys/parsoid.js (hunk 1 above)
Comment (Pchelolo) [reply sys/parsoid.js]: Still need. this is for non-latest revision edits.

File: sys/parsoid.js (hunk 2 above)
Comment (Pchelolo) [reply sys/parsoid.js]: We will not resurrect it ever afaik after talking to Subbu. I will remove it from this PR though, you're right.

File: sys/parsoid.js (hunk 3 above)
Comment (Pchelolo) [reply sys/parsoid.js]: Ulala... let's not. Judging by the history of the whole 'decodeBody' magic will have to go. It's all about zipping up the content that we don't do anymore. Plus, after split this line will go away too.

File: sys/parsoid.js (hunk 4 above)
Comment (Pchelolo) [reply sys/parsoid.js]: Idem

**Comment by d00rman** [comment]
Alright! Let's try this stuff out :)
